modin/engines/base/io/text/csv_dispatcher.py

```diff
@@ -83,115 +91,141 @@ def _read(cls, filepath_or_buffer, **kwargs):
         partition_kwargs = dict(
             kwargs,
+            fname=filepath_or_buffer,
+            num_splits=num_splits,
             header=None,
             names=names,
             skipfooter=0,
-            skiprows=None,
+            skiprows=1 if encoding is not None else None,
             parse_dates=parse_dates,
             usecols=usecols,
+            nrows=None,
+            compression=compression_type,
+            index_col=index_col,
         )
-        encoding = kwargs.get("encoding", None)
-        quotechar = kwargs.get("quotechar", '"').encode(
-            encoding if encoding is not None else "UTF-8"
-        )
-        is_quoting = kwargs.get("quoting", "") != csv.QUOTE_NONE
+
         with cls.file_open(filepath_or_buffer, "rb", compression_type) as f:
-            # Skip the header since we already have the header information and skip the
-            # rows we are told to skip.
-            if isinstance(skiprows, int) or skiprows is None:
-                if skiprows is None:
-                    skiprows = 0
-                header = kwargs.get("header", "infer")
-                if header == "infer" and kwargs.get("names", None) is None:
-                    skiprows += 1
-                elif isinstance(header, int):
-                    skiprows += header + 1
-                elif hasattr(header, "__iter__") and not isinstance(header, str):
-                    skiprows += max(header) + 1
-            if kwargs.get("encoding", None) is not None:
-                partition_kwargs["skiprows"] = 1
-            # Launch tasks to read partitions
-            partition_ids = []
-            index_ids = []
-            dtypes_ids = []
-            # Max number of partitions available
-            num_partitions = NPartitions.get()
-            # This is the number of splits for the columns
-            num_splits = min(len(column_names), num_partitions)
-            # Metadata
-            column_chunksize = compute_chunksize(empty_pd_df, num_splits, axis=1)
-            if column_chunksize > len(column_names):
-                column_widths = [len(column_names)]
-                # This prevents us from unnecessarily serializing a bunch of empty
-                # objects.
-                num_splits = 1
-            else:
-                column_widths = [
-                    column_chunksize
-                    if len(column_names) > (column_chunksize * (i + 1))
-                    else 0
-                    if len(column_names) < (column_chunksize * i)
-                    else len(column_names) - (column_chunksize * i)
-                    for i in range(num_splits)
-                ]
-
-            args = {
-                "fname": filepath_or_buffer,
-                "num_splits": num_splits,
-                **partition_kwargs,
-            }
+            rows_skipper = cls.rows_skipper_builder(f, quotechar, is_quoting=is_quoting)
+            rows_skipper(skiprows)
```

Comment:
Why not directly call `_read_rows`?

Reply:
There is no special reason for this, is is just to make `_read` function lighter(in the future `rows_skipper` can be used several times).